src/SelfService/Application/KafkaTopicApplicationService.cs

```diff
@@ -24,6 +26,205 @@ IKafkaTopicRepository kafkaTopicRepository
         _kafkaTopicRepository = kafkaTopicRepository;
     }
 
+    public async Task CheckIfCanRequestContract(
```

Comment:
This naming is confusing to me.
Yes we are making a request to create a contract, but in everyday speak 'requesting' would mean asking for something.

I would think `canRequestContract` to check if a user is allowed to read a contract.
That does not seem to be the case here.

Reply:
I dont really care just give me the name you want. The function throws exceptions on failure